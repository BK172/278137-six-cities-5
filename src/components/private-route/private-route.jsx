import React from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";

const PrivateRoute = ({
  render,
  path,
  exact,
  isAuth,
  redirectURL,
}) => {
  return (
    <Route
      path={path}
      exact={exact}
      render={(routeProps) => {
        return (
          isAuth
            ? render(routeProps)
            : <Redirect to={redirectURL} />
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  isAuth: PropTypes.bool.isRequired,
  redirectURL: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
};

export default PrivateRoute;
