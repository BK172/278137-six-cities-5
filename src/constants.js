export const ResponseType = {
  SUCCESS: `SUCCESS`,
  ERROR: `ERROR`,
};

export const HttpCode = {
  UNAUTHORIZED: 401,
  SUCCESS: 200,
};

export const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
  ROOM: `/offer/:id`,
  ERROR: `/error`,
};

export const APIRoute = {
  OFFERS: `/hotels`,
  NEARBY: `/nearby`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  REVIEWS: `/comments`,
  FAVORITE: `/favorite`,
};

export const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const SortingTypesNames = {
  POPULAR: `popular`,
  TO_HIGH: `to-high`,
  TO_LOW: `to-low`,
  TOP_RATED: `top-rated`,
};

export const SortingTypes = {
  [SortingTypesNames.POPULAR]: `Popular`,
  [SortingTypesNames.TO_HIGH]: `Price: low to high`,
  [SortingTypesNames.TO_LOW]: `Price: high to low`,
  [SortingTypesNames.TOP_RATED]: `Top rated first`,
};

export const INITIAL_SORTING_TYPE = SortingTypesNames.POPULAR;

export const MARKUP_CITIES_LIST = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

export const MAX_PHOTOS_COUNT = 6;

export const REVIEW_DATE_FORMAT = `MMMM YYYY`;
export const REVIEW_RATING_ELEMENT_WIDTH = 20;
export const MAX_REVIEWS_ON_PAGE = 10;

export const ReviewFormTextAreaLength = {
  MIN: 50,
  MAX: 300,
};

export const ReviewFormRatings = [
  {mark: `5`, title: `perfect`},
  {mark: `4`, title: `good`},
  {mark: `3`, title: `not bad`},
  {mark: `2`, title: `badly`},
  {mark: `1`, title: `terribly`},
];

export const OfferType = {
  MAIN: `MAIN`,
  ROOM: `ROOM`,
  FAVORITE: `FAVORITE`,
};

export const OfferClasses = {
  MAIN: {
    article: `cities__place-card`,
    wrapper: `cities__image-wrapper`,
  },
  ROOM: {
    article: `near-places__card`,
    wrapper: `near-places__image-wrapper`,
  },
  FAVORITE: {
    article: `favorites__card`,
    wrapper: `favorites__image-wrapper`,
    info: `favorites__card-info`,
  },
};

export const OfferImgShapes = {
  MAIN: {
    width: 260,
    height: 200,
  },
  ROOM: {
    width: 260,
    height: 200,
  },
  FAVORITE: {
    width: 150,
    height: 110,
  },
};

export const MapType = {
  MAIN: `MAIN`,
  ROOM: `ROOM`,
};

export const MapClasses = {
  MAIN: `cities__map`,
  ROOM: `property__map`,
};

export const BookmarkBtnType = {
  MAIN: `MAIN`,
  ROOM: `ROOM`,
};

export const BookmarkBtnClasses = {
  MAIN: {
    btn: `place-card__bookmark-button button`,
    btnActive: `place-card__bookmark-button--active`,
    icon: `place-card__bookmark-icon`,
  },
  ROOM: {
    btn: `property__bookmark-button button`,
    btnActive: `property__bookmark-button--active`,
    icon: `property__bookmark-icon`,
  },
};

export const BookmarkBtnShapes = {
  MAIN: {
    width: 18,
    height: 19,
  },
  ROOM: {
    width: 31,
    height: 33,
  },
};

export const MAP_TILE_LAYER = `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`;
export const MAP_TILE_LAYER_ATTRIBUTION = `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`;
