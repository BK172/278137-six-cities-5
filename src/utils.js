import _ from "lodash";

const MarkupCitiesList = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

export const ResponseType = {
  SUCCESS: `SUCCESS`,
  ERROR: `ERROR`,
};

export const HttpCode = {
  UNAUTHORIZED: 401,
};

export const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  ROOM: `/offer/:id`,
};

export const APIRoute = {
  OFFERS: `/hotels`,
  LOGIN: `/login`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const getElementWidthByRating = (rating) => Math.round(rating) * 20;

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const initialSortingType = `popular`;
export const SortingTypes = {
  'popular': `Popular`,
  'to-high': `Price: low to high`,
  'to-low': `Price: high to low`,
  'top-rated': `Top rated first`
};

export const orderCitiesByList = (cities) => {
  const orderedCities = [];

  MarkupCitiesList.forEach((cityName) => {
    const city = cities.find((city) => city.name === cityName);
    city && orderedCities.push(city);
  });

  return orderedCities;
};

export const getCitiesFromOffersList = (offers) => {
  const cities = offers.map((offer) => offer.city);
  return _.uniqWith(cities, _.isEqual);
};

export const citiesAdapter = (city) => {
  return {
    name: city.name,
    coordinates: [city.location.latitude, city.location.longitude],
    zoom: city.location.zoom,
  };
};

export const offersAdapter = (offer) => {
  return {
    bedrooms: offer.bedrooms,
    city: citiesAdapter(offer.city),
    description: offer.description,
    facilities: offer.goods,
    owner: {
      id: offer.host.id,
      avatar: offer.host.avatar_url,
      name: offer.host.name,
      super: offer.host.is_pro,
    },
    offerId: offer.id,
    photos: offer.images,
    favorite: offer.is_favorite,
    premium: offer.is_premium,
    coordinates: [offer.location.latitude, offer.location.longitude],
    zoom: offer.location.zoom,
    guests: offer.max_adults,
    image: offer.preview_image,
    price: offer.price,
    rating: offer.rating,
    title: offer.title,
    type: offer.type,
  };
};
