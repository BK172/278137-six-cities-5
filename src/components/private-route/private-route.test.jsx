import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import configureStore from "redux-mock-store";
import {PrivateRoute} from "./private-route";
import {makeInitialStateMock} from "../../utils";
import {AuthStatus} from "../../constants";

describe(`Should PrivateRoute render correctly`, () => {
  const store = configureStore()(makeInitialStateMock());

  it(`Should PrivateRoute render correctly authStatus={AuthStatus.NO_AUTH}`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <PrivateRoute
                authStatus={AuthStatus.NO_AUTH}
                exact={true}
                path={``}
                render={()=>{}}
              />
            </BrowserRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Should PrivateRoute render correctly authStatus={AuthStatus.AUTH}`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <PrivateRoute
                authStatus={AuthStatus.AUTH}
                exact={true}
                path={``}
                render={()=>{}}
              />
            </BrowserRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
