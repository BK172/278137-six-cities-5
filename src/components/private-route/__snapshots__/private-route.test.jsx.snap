// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PrivateRoute render correctly Should PrivateRoute render correctly authStatus={AuthStatus.AUTH} 1`] = `null`;

exports[`Should PrivateRoute render correctly Should PrivateRoute render correctly authStatus={AuthStatus.NO_AUTH} 1`] = `null`;
